components/camel-olingo4/camel-olingo4-component/src/main/java/org/apache/camel/component/olingo4/Olingo4Consumer.java

```diff
@@ -82,7 +83,15 @@ public void onCanceled() {
                 throw error[0];
             }
 
-            return ApiConsumerHelper.getResultsProcessed(this, result[0], isSplitResult());
+            //
+            // Allow consumer idle properties to properly handle an empty polling response
+            //
+            int processed = ApiConsumerHelper.getResultsProcessed(this, result[0], isSplitResult());
```

Comment:
We could maybe moved this where `return processed` is so its only evaluated if needed

Reply:
Fixed.